### PR TITLE
Add draft of python side configurability

### PR DESCRIPTION
**Pull request by Carreau**
@KesterTong Pull agains your fork. 

Making Python side config manager configurable.
(in progress)

### CODE REVIEW

**Comment by Carreau** [comment]
OK, kinda-working, there is the small issue in passing config around as clientside content manager should get the root of path removed, while the "On disk" mixed one should not. 

So it's still hardcoded, but working.

**Comment by KesterTong** [comment]
Got it.  Seems like the best solution is to modify the JavaScript mixed Contents class, to have the option to pass the full path on to the Contents instance.  This option would be set true for the local Contents instance and false for the Drive instance.

**Comment by KesterTong** [comment]
I guess that is what `stripjs` is meant for?

**Comment by Carreau** [comment]
Yep, that was what `stripjs` is for, haven't just finished implementing the logic.

**Comment by Carreau** [comment]
Ok, that should work.

Now cleanup and docs.

File: jupyterdrive/mixednbmanager.py [hunk 1]
```diff
@@ -4,68 +4,138 @@
...
+        kwargs.update({'parent':self})
+        for scheme in self.filesystem_scheme:
+            manager_class = import_item(scheme['contents'])
+            self.managers[scheme['root']] = manager_class(**kwargs)
+
+
+    def path_dispatch1(method):
+        def _wrapper_method(self, path, *args, **kwargs):
+            path = path.strip('/')
+            (sentinel, *_path) = path.split('/')
```
Comment (KesterTong) [inline jupyterdrive/mixednbmanager.py]: I'm getting a syntax error at this line.  Is this a Python 2 vs 3 thing?

Comment (KesterTong) [inline jupyterdrive/mixednbmanager.py]: yeah it is, extended sequence unpacking is Python 3 only.

**Comment by KesterTong** [comment]
Everything is working except the root in tree view, i.e. `localhost:8888/tree`.  Then I get the following stack trace

```
    Traceback (most recent call last):
      File "/usr/local/google/home/kestert/ip/local/lib/python2.7/site-packages/tornado/web.py", line 1332, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/usr/local/google/home/kestert/ip/local/lib/python2.7/site-packages/tornado/web.py", line 2607, in wrapper
        return method(self, *args, **kwargs)
      File "/usr/local/google/home/kestert/ipython/IPython/html/tree/handlers.py", line 51, in get
        elif cm.file_exists(path):
      File "/usr/local/google/home/kestert/jupyter-drive/jupyterdrive/mixednbmanager.py", line 80, in _wrapper_method
        return method(self, path=path)
      File "/usr/local/google/home/kestert/jupyter-drive/jupyterdrive/mixednbmanager.py", line 103, in file_exists
        raise NotImplementedError('NotImplementedError')
    NotImplementedError: NotImplementedError
```

File: jupyterdrive/mixednbmanager.py (hunk 1 above)
Comment (KesterTong) [inline jupyterdrive/mixednbmanager.py]: It works for me when I use

```
_path = path.split('/')
sentinel = _path.pop(0)
```

Maybe you can think of something more elegant that still works in Python 2.7?

**Comment by Carreau** [comment]
Modified as you asked.

This is a PR agains your repo, not jpyter/jupyterdrive. You can also merge it and push change that you need directly if needed.

Once merge it will update https://github.com/jupyter/jupyter-drive/pull/79 that then we can merge into original repo.

**Comment by Carreau** [comment]
I don't see the not implemented error on /tree ... can you debug the requested path ?

**Pull request merged**